doc: update documents on using kcephfs (#30626)
ceph/ceph: doc/man/8/mount.ceph.rst

  for autodiscovery of monitor addresses and auth secrets. The default is
  to use the standard search path for ceph.conf files.

:command:`recover_session=<no|clean>`
  Set auto reconnect mode in the case where the client is blacklisted. The
  available modes are ``no`` and ``clean``. The default is ``no``.

>   - ``no``: never attempt to reconnect when client detects that it has been blacklisted. Operations will generally fail after being blacklisted.

Reviewer: Another small change:
```suggestion
  - ``no``: Blacklisted clients will not attempt to reconnect and their operations will fail too.
```
